ADR/0039-workspace-deprecation.md

```diff
@@ -0,0 +1,104 @@
+# 39. Workspace Deprecation
+
+Some of the text was taken from - https://github.com/konflux-ci/architecture/pull/187
+
+## Status
+
+WiP
+
+## Context
+
+The purpose of this ADR is to revisit the workspace concept, understand the purpose
+of it, and offer an alternative implementation based on native Kubernetes APIs
+and other successful cloud native open source projects.
+
+We inherited the "workspace" concept from our early days when we expected to be integrated with the [KCP] project. Long ago (over a year at the time of this writing), we decided to remove [KCP] from the architecture of the Konflux project.
+
+In order to support multi-cluster deployment, [kubesaw] was used instead of KCP.
+
+Today, Workspaces are an important concept in the Konflux user experience. The user onboards into a workspace of their own. A team member of theirs may add them to a team workspace. As a platform administrator, if you hear about a user having trouble, you'll ask them what workspace they're working in.
+
+A Workspace is an abstraction of a namespace. Each workspace has a single namespace
+which backs it.
+
+Workspaces are required by [kubesaw] for providing a transparent multi-cluster communication for us. A user's workspace may be allocated to one cluster or another - but our proxy layer keeps the user blissfully unaware of this. Workspaces can be placed on clusters either automatically (using [kubesaw] mechanisms) or manually by a platform administrator.
+
+Workspace roles give a way to describe who has what rights in the workspace.
+
+Workspaces happen to be our mechanism for managing quota - a fact inherited from the [kubesaw] stack.
+
+### Some changes
+
+Some relevant changes that have occurred or are ongoing.
+
+* We want to make Konflux a successful open source projects that fits into the
+cloud native apps ecosystem. We want everything to work on vanilla Kubernetes - no OpenShift required.
+
+* We had our [first upstream Konflux release](https://github.com/konflux-ci/konflux-ci) which doesn't use [kubesaw] or relying
+on the Workspace abstraction, and is deployable on vanilla Kubernetes.
+
+* We made a decision (now long ago) to drop [KCP] from our architecture, which initially introduced the workspace concept. Both transparent multi-cluster as a feature and workspaces as a concept came from KCP.
+
+* We made the decision (at Red Hat) to stop trying to offer a managed service interface for Konflux. The user signup, waitlist, and workspace provisioning interfaces made a lot of sense there.
+
+## Decision
+
+1. We will remove the Workspace abstraction from Konflux and use standard Kubernetes namespaces
+directly for providing a place for users to run their pipelines and store their configurations
+and secrets.
+
+2. Konflux won't provide its own policy engine for enforcing policies related to
+creating/deleting namespaces. It will instead recommend user to use existing open source projects
```

Comment:
Will like clarification on this. Are we only going to suggest some policy engines but not the policies themself. My understanding discussing with @brianwcook is the expectations was there was a minimum default way to managed namespaces upstream and each deployment will decide if they need more. The simplest way I can see we can suggest upstream is instructions to create a namespace and a rolebinding inside to a existing ClusterRole. But I think others are expecting more while this lines asserts "Konflux won't provide" a specific solution

Reply:
When using role aggregation, mentioned in section 8, even a role binding to a Konflux specific role shouldn't be created.

Reply:
If the are no objections I'll close this thread.